libsoftwarecontainer/src/gateway/devicenode/devicenode.cpp

```diff
@@ -0,0 +1,94 @@
+/*
+ * Copyright (C) 2017 Pelagicore AB
+ *
+ * Permission to use, copy, modify, and/or distribute this software for
+ * any purpose with or without fee is hereby granted, provided that the
+ * above copyright notice and this permission notice appear in all copies.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL
+ * WARRANTIES WITH REGARD TO THIS SOFTWARE INCLUDING ALL IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS. IN NO EVENT SHALL THE AUTHOR
+ * BE LIABLE FOR ANY SPECIAL, DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES
+ * OR ANY DAMAGES WHATSOEVER RESULTING FROM LOSS OF USE, DATA OR PROFITS,
+ * WHETHER IN AN ACTION OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION,
+ * ARISING OUT OF OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS
+ * SOFTWARE.
+ *
+ * For further information see LICENSE
+ */
+
+#include <sys/stat.h>
+
+#include "devicenode.h"
+#include "functionjob.h"
+
+namespace softwarecontainer {
+
+Device::Device(std::string _name, int _mode, bool _configured) :
+        name(_name),
+        mode(_mode),
+        isConfigured(_configured)
+{
+}
+
+Device::Device(const Device &dev) :
+        name(dev.name),
+        mode(dev.mode),
+        isConfigured(dev.isConfigured)
+{
+}
+
+bool Device::parse(const json_t *element)
```

Comment:
I would prefer if this function was a static factory function, so that it creates a Device object on successful parse.